### PR TITLE
coordinate parser extension

### PR DESCRIPTION
**Pull request by Mike-3**
paste in all coordinates formats possibile
utm widget format changed (default format) 
dec, dmm, dms format splitted

### CODE REVIEW

**Comment by S-Man42** [comment]
Please resolve conflicts

File: lib/logic/tools/coords/converter/open_location_code.dart [hunk 1]
```diff
@@ -302,62 +302,67 @@ String latLonToOpenLocationCode(LatLng coords, {int codeLength = pairCodeLength}
...
-  // Decode the paired digits.
-  for (var i = 0; i < digits; i += 2) {
-    normalLat += codeAlphabet.indexOf(code[i]) * pv;
-    normalLng += codeAlphabet.indexOf(code[i + 1]) * pv;
-    if (i < digits - 2) {
-      pv = pv ~/ encodingBase;
+  try{
+    if (!isFull(code)) {
+      return null;
+      //throw ArgumentError('coords_formatconverter_openlocationcode_novalidcode');
```
Comment (S-Man42) [inline lib/logic/tools/coords/converter/open_location_code.dart:308]: Please remove this comment

File: lib/logic/tools/coords/converter/utm.dart [hunk 2]
```diff
@@ -152,7 +152,7 @@ LatLng parseUTM(String input, Ellipsoid ells) {
     _northingString = match.group(4);
   }
   if (matches.length == 0) {
-    regExp = RegExp(r'^\s*(\d+)\s?(['+ latZones + r'])\s?m\s?([EW])\s?([0-9\.]+)\s?m\s?([NS])\s?([0-9\.]+)\s*$');
+    regExp = RegExp(r'^\s*(\d+)\s?([T])\s*m?\s*([E|W])\s*([0-9\.]+)\s*m?\s*([N|S])\s?([0-9\.]+)\s*$');
```
Comment (S-Man42) [inline lib/logic/tools/coords/converter/utm.dart:155]: The change from "latZones" to T is confusing me... latZones contain much more letters than just a T. Or do I misunderstand something here?

**Review by S-Man42 — COMMENTED** [review]
Seems to work quite well. Great job! I really feared this review before...

However: Some minor remarks. 

And I was not able to proof all widget changes, I guess we need a huge test in a test version for that.

File: assets/i18n/en.json [hunk 3]
```diff
@@ -986,22 +986,23 @@
...
-  "coords_common_centerpoint": "Center Point",
-  "coords_common_intersection": "Intersection",
-  "coords_common_clipboard_nocoordsfound":  "No coordinates found within the paste data.",
-  "coords_common_location_permissiondenied":  "Access to position data denied.",
-  "coords_common_location_lowaccuracy":  "WARNING: Low accuracy (%s1 m)",
+  "coords_common_centerpoint" : "Center Point",
+  "coords_common_intersection" : "Intersection",
+  "coords_common_clipboard_nocoordsfound" : "No coordinates found within the paste data.",
+  "coords_common_clipboard_recognizedcoordformats" : "Recognized coordinate formats",
+  "coords_common_location_permissiondenied" :  "Access to position data denied.",
```
[the remark this answers is not in the document]
Comment (Mike-3) [reply assets/i18n/en.json:993]: Good question. The "change" came from a merge with 1.5.0. But actually these texts were already there and are also needed.
I don't know why those show up as a change.

File: lib/logic/tools/coords/converter/gauss_krueger.dart [hunk 4]
```diff
@@ -126,18 +126,34 @@ String latLonToGaussKruegerString(LatLng coords, int gaussKruegerCode, Ellipsoid
 }
 
 LatLng parseGaussKrueger(String input, Ellipsoid ells, {gaussKruegerCode: 1}) {
-  RegExp regExp = RegExp(r'^\s*([\-0-9\.]+)(\s*,\s*|\s+)([\-0-9\.]+)\s*$');
+  RegExp regExp = RegExp(r'^\s*([0-9\.]+)(\s*\,\s*|\s+)([0-9\.]+)\s*$');
```
[the remark this answers is not in the document]
Comment (Mike-3) [reply lib/logic/tools/coords/converter/gauss_krueger.dart:129]: Ok. I change this back.

[the remark this answers is not in the document]
Comment (Mike-3) [reply lib/logic/tools/coords/converter/gauss_krueger.dart:129]: But the minimum for the input field was 0. I'll change that too.

File: lib/logic/tools/coords/converter/maidenhead.dart [hunk 5]
```diff
@@ -10,26 +12,31 @@ LatLng maidenheadToLatLon (String maidenhead) {
...
+        lon += int.tryParse(maidenhead[i]) * reslon;
+        lat += int.tryParse(maidenhead[i + 1]) * reslat;
+        ++res;
+      }
     }
+  } catch(e) {
+    return null;
   }
+  if (lat == 0.0 && lon == 0.0)
+    return null;
```
[the remark this answers is not in the document]
Comment (Mike-3) [reply lib/logic/tools/coords/converter/maidenhead.dart:39]: I did that in order to detect an invalid input. In the case of invalid inputs, lat / lon 0.0 also comes out.
I'm trying to find another way of recognizing invalid inputs.
NULL is the recognition that no coordinate could be recognized. With all coordinate formats.

[the remark this answers is not in the document]
Comment (S-Man42) [reply lib/logic/tools/coords/converter/maidenhead.dart:39]: Two Options: You return a Map

{status: ok, value: latlon} or {status: error} 

Or you simply throw an exception...

File: lib/logic/tools/coords/converter/utm.dart (hunk 2 above)
Comment (Mike-3) [reply lib/logic/tools/coords/converter/utm.dart:155]: ok

File: lib/logic/tools/coords/converter/maidenhead.dart (hunk 5 above)
[the remark this answers is not in the document]
Comment (Mike-3) [reply lib/logic/tools/coords/converter/maidenhead.dart:39]: The problem ist not the return value. 
The problem is the calculation of the coordinates. If invalid values are given when determining the coordinates, this type of calculation always results in 0. As no matter what I give the part to parse, I would always have the valid coordinates lan / lot 0.0.

**Pull request merged**